--- rpm-4.8.0/scripts/perl.req.orig	2011-11-08 10:22:35.339068490 -0500
+++ rpm-4.8.0/scripts/perl.req	2011-11-08 10:23:25.804976974 -0500
@@ -98,8 +98,9 @@
     if ( m/^.*\Wq[qxwr]?\s*([\{\(\[#|\/])[^})\]#|\/]*$/ && ! m/^\s*(require|use)\s/ ) {
       $tag = $1;
       $tag =~ tr/{\(\[\#|\//})]#|\//;
+      $tag = quotemeta($tag);
       while (<FILE>) {
-        ( $_ =~ m/\}/ ) && last;
+        ( $_ =~ m/$tag/ ) && last;
       }
     }
 
